src/continuum_opacity/opacity_H.jl

```diff
@@ -0,0 +1,237 @@
+- `ne::Flt`: the number density of free electrons.
+- `ν::Flt`: frequency in Hz
+- `ρ::Flt`: mass density in g/cm³
+- `T::Flt`: temperature in K
+
+# Notes
+This is taken from equation 8.13 of Gray (2005). This uses a polynomial to Bell & Berrington (1987)
+tabulated values for α_ff(H⁻), which implicitly includes the correction for stimulated emission.
+The polynomial fits α_ff(H⁻) in the range 2520 K ≤ T ≤ 10080 K and 2604 Å ≤ λ ≤ 113918 Å. According
+to the book, the polynomial fit to the tabulated data typically has 1% precision. We find that at
+worst, the discrepancy never exceeds 2.25%.
+
+From equations 8.13, 8.18, and 8.19 of Gray (2005), the free-free opacity from H⁻ is given by:
+                              n(H I)          n(H I) + n(H II)
+   κ_ν = α_ff(H⁻) * Pₑ * ------------------ * -----------------
+                          n(H I) + n(H II)            ρ
+This can be rewritten as: κ_ν = α_ff(H⁻) * Pₑ * n(H I) / ρ
+
+Based on Section 5.3 from Kurucz (1970), I'm fairly confident that the "more correct" version of
+the opacity equation should actually read  κ_ν = α_ff(H⁻) * Pₑ * n(H I, n = 1) / ρ, and that the
+form provided by Gray (2005) implicitly assumes that n(H I, n = 1) ~ n(H I). This seems to be a 
+prettygood assumption given the tabulated values of the partition function in Table D.2 of Gray 
+(2005). From the boltzmann equation:
+     n(H I, n = 1) = n(H I)*gₙ₌₁/u(T)*exp(-Eₙ₌₁/(k*T)) = n(H I) * 2/u(T)*exp(0) = n(H I) * 2/u(T).
+According to Table D.2 of Grey, this approximation only introduces a ≤0.5% overestimate in 
+n(H I, n = 1) over the temperature range where the polynomial is valid.
+
+We also considered the polynomial fit in Section 5.3 from Kurucz (1970). Unfortunately, it seems
+wrong (it gives lots of negative numbers).
+"""
+function Hminus_ff(nH_I_div_partition::Flt, ne::Flt, ν::Flt, ρ::Flt,
+                   T::Flt) where {Flt<:AbstractFloat}
+    λ = c_cgs*1e8/ν # in Angstroms
+    # we need to somehow factor out this bounds checking
+    if (λ < 2604.0) || (λ > 113918.0)
+        throw(DomainError(ν, "The wavelength must lie in the interval [2604 Å, 113918 Å]"))
+    elseif (T < 2520.0) || (T > 10080.0)
+        throw(DomainError(T, "The temperature must lie in the interval [2520 K, 10080 K]"))
+    end
+
+    logλ = log10(λ)
+    log2λ = logλ * logλ
+    log3λ = log2λ * logλ
+    log4λ = log3λ * logλ
+
+    f0 =  -2.2763 -   1.6850 * logλ +  0.76661 * log2λ -  0.053346 * log3λ
+    f1 = +15.2827 -   9.2846 * logλ +  1.99381 * log2λ -  0.142631 * log3λ
+    f2 = -197.789 + 190.266  * logλ - 67.9775  * log2λ + 10.6913   * log3λ - 0.625151 * log4λ
+
+    logθ = log10(5040.0/T)
+    αff_H⁻ = 1e-26 * 10.0^(f0 + f1 * logθ + f2 * (logθ * logθ))
+    # Pₑ * α_ff(H⁻) gives the absorption coefficient in units of cm² per ground state H I atom
+    Pₑ = ne * kboltz_cgs * T
+
+    # in this case, I'm going to account for the fact that n(H I, n=1) might be slightly
```

Comment:
In what case?

Reply:
Good catch. There was originally an if-statement here. I'll fix it.